components/selectors/bloom.rs

```diff
@@ -179,9 +136,84 @@ impl BloomFilter {
     }
 }
 
-#[inline]
-fn full(slot: &u8) -> bool {
-    *slot == 0xff
+pub trait BloomStorage : Clone + Default {
+    fn slot_is_empty(&self, index: usize) -> bool;
+    fn adjust_slot(&mut self, index: usize, increment: bool);
```

Comment:
Could you mention what are these functions expected to do?

I found `adjust` to be a little not too descriptive... But I can't think of a better name that isn't super-long, so, maybe something simple like:

```
Either increment or decrement the slot at `index`.
```

Would be helpful.